flaml/data.py

```diff
@@ -253,13 +265,23 @@ def transform(self, X):
         X = X.copy()
         if isinstance(X, pd.DataFrame):
             cat_columns, num_columns, datetime_columns = self._cat_columns, \
-                self._num_columns, self._datetime_columns
-            X = X[cat_columns + num_columns].copy()
+                                                         self._num_columns, self._datetime_columns
             if datetime_columns:
-                for dt_column in datetime_columns:
-                    X[dt_column] = X[dt_column].map(datetime.toordinal)
+                for column in datetime_columns:
+                    tmp_dt = X[column].dt
+                    new_columns_dict = {f'year_{column}':      tmp_dt.year, f'month_{column}': tmp_dt.month,
+                                        f'day_{column}':       tmp_dt.day, f'hour_{column}': tmp_dt.hour,
+                                        f'minute_{column}':    tmp_dt.minute, f'second_{column}': tmp_dt.second,
+                                        f'dayofweek_{column}': tmp_dt.dayofweek,
+                                        f'dayofyear_{column}': tmp_dt.dayofyear,
+                                        f'quarter_{column}':   tmp_dt.quarter}
+                    for new_col_name in new_columns_dict.keys():
+                        if new_col_name not in X.columns:
```

Comment:
also need to check if the new_col_name is in num_columns in case they are dropped